store/v2/multi/snapshot.go

```diff
@@ -0,0 +1,167 @@
+					Name: string(sKey),
+				},
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		iter := subStore.Iterator(nil, nil)
+		for ; iter.Valid(); iter.Next() {
+			err = protoWriter.WriteMsg(&snapshottypes.SnapshotItem{
+				Item: &snapshottypes.SnapshotItem_KV{
+					KV: &snapshottypes.SnapshotKVItem{
+						Key:   iter.Key(),
+						Value: iter.Value(),
+					},
+				},
+			})
+			if err != nil {
+				return err
+			}
+		}
+
+		err = iter.Close()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Restore implements snapshottypes.Snapshotter.
+func (rs *Store) Restore(
+	height uint64, format uint32, protoReader protoio.Reader,
+) (snapshottypes.SnapshotItem, error) {
+	if err := snapshots.ValidRestoreHeight(format, height); err != nil {
+		return snapshottypes.SnapshotItem{}, err
+	}
+
+	if rs.LastCommitID().Version != 0 {
+		return snapshottypes.SnapshotItem{}, sdkerrors.Wrapf(sdkerrors.ErrLogic, "cannot restore snapshot for non empty store at height %v", height)
+	}
+
+	var subStore *substore
+	var storeSchemaReceived = false
+
+	for {
+		snapshotItem := &snapshottypes.SnapshotItem{}
+		err := protoReader.ReadMsg(snapshotItem)
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return snapshottypes.SnapshotItem{}, sdkerrors.Wrap(err, "invalid protobuf message")
+		}
+
+		switch item := snapshotItem.Item.(type) {
+		case *snapshottypes.SnapshotItem_Schema:
+			receivedStoreSchema := make(StoreSchema, len(item.Schema.GetKeys()))
+			storeSchemaReceived = true
+			for _, sKey := range item.Schema.GetKeys() {
+				receivedStoreSchema[string(sKey)] = types.StoreTypePersistent
+			}
+
+			if !rs.schema.equal(receivedStoreSchema) {
+				return snapshottypes.SnapshotItem{}, sdkerrors.Wrap(sdkerrors.ErrLogic, "received store schema is not matched with app schema empty")
+			}
+
+		case *snapshottypes.SnapshotItem_Store:
+			storeName := item.Store.GetName()
+			// checking the store schema is received or not
+			if !storeSchemaReceived {
+				return snapshottypes.SnapshotItem{}, sdkerrors.Wrapf(sdkerrors.ErrLogic, "received store name before store schema %s", storeName)
+			}
+			// checking the store schema exists or not
+			if _, has := rs.schema[storeName]; !has {
+				return snapshottypes.SnapshotItem{}, sdkerrors.Wrapf(sdkerrors.ErrLogic, "store is missing from schema %s", storeName)
+			}
+
+			// get the substore
+			subStore, err = rs.getSubstore(storeName)
+			if err != nil {
+				return snapshottypes.SnapshotItem{}, sdkerrors.Wrap(err, fmt.Sprintf("error while getting the substore for key %s", storeName))
+			}
+
+		case *snapshottypes.SnapshotItem_KV:
+			if subStore == nil {
+				return snapshottypes.SnapshotItem{}, sdkerrors.Wrap(sdkerrors.ErrLogic, "received KV Item before store item")
+			}
+			// update the key/value SMT.Store
+			subStore.Set(item.KV.Key, item.KV.Value)
+
+		default:
+			return snapshottypes.SnapshotItem{}, sdkerrors.Wrapf(sdkerrors.ErrLogic, "unknown snapshot item %T", item)
```

Comment:
When meet an unknown item, we should treat it as success, and return that unknown item to snapshot manager, because that could be the beginning of the next extension snapshotter.

Reply:
I fixed a similar issue in rootmulti store's `Restore` implementation: https://github.com/cosmos/cosmos-sdk/pull/11286